### DISCUSSION TITLE
Azure Performance Issues

### DISCUSSION THREAD

**Discussion by MattHartz** [IEvangelist/azure-cosmos-dotnet-repository#421] [Q&A]
Hi all,

I know this isn't an Azure board but I have been noticing some odd performance issues when working in Azure. 

I've been investigating some performance issues when using this library and cosmosdb. Using Azure and application insights i noticed some odd performance behavior. It seems that for whatever reason there's a large gap occasionally in requests. I'm not understanding why? I am using the "specification" pattern for querying my cosmos db container, using the "OffsetByPageNumberSpecification" type. I assume this specification queries cosmos twice, once for the metadata of how many documents are in the container and the 2nd one actually queries the documents.

Does anyone have any idea what might cause a lag between these 2 queries?
![image](https://github.com/IEvangelist/azure-cosmos-dotnet-repository/assets/16110147/f4c97893-6689-4fb1-9aeb-fcb8daf4e1f9)
in the above API there was a whole second of waiting in between these calls

Some additional context, this is a projection container that is managed by the ChangeFeedProcessorHostedService stuff. 

Lastly, this is fairly common for a large lag, probably about 5-10% of the time. This lag isn't usually a second though, but sometimes hundreds of milliseconds

**Comment by IEvangelist** [comment]
Let's tag @mumby0168 and see if he has any quick answers.

**Reply by MattHartz** [reply]
Thanks @IEvangelist 

Yeah i've been battling some stuff trying to find out what is happening in Azure. Granted i think there is still a lul in between those 2 calls, but it's greatly affecting API performance. Only thing i can think is occasional CPU performance... but even it it's best cases there's like 80ms added
